server/backend/api/app/exception_handlers.py

```diff
@@ -23,9 +23,24 @@ async def requests_http_error_handler(request: Request, exc: HTTPError):
     port = getattr(getattr(request, "client", None), "port", None)
     url = f"{request.url.path}?{request.query_params}" if request.query_params else request.url.path
     response_text = json.loads(exc.response.text)
+
+    failureCode = None
+    if "failureCode" in response_text:
+        failureCode = response_text["failureCode"]
+    elif "errors" in response_text and "error" in response_text["errors"][0]:
+        # this is the error format for gc notify
+        failureCode = response_text["errors"][0]["error"]
+
+    error_message = None
+    if "message" in response_text:
+        error_message = response_text["message"]
+    elif "errors" in response_text and "message" in response_text["errors"][0]:
+        # this is the error format for gc notify
+        error_message = response_text["errors"][0]["message"]
+
```

Comment:
Just a note here.
As you probably **will remove the "router" for GC notify** (for your testing purpose) later, I am thinking adding gc notify modification in "exception_handlers.py" here probably has no point.

I am quite positive (but not 100% sure), the exception_handlers.py is for **FastAPI**, so the exception you get from gc notify probably only will be propagated and caught (by **exception_handlers**) until reaching the "**router**". So if the router will be removed, you might need to handle exception in the email sending code itself or create a **custom exception** to throw for CRUD code to handle email sending errors.

Reply:
Hi Ian, I think we use this general exception handler in all crud methods, so I changed in this general place, do you feel I need to handle the email error separately, or this should be okay?

<img width="750" alt="Screen Shot 2023-08-15 at 10 02 26 AM" src="https://github.com/bcgov/nr-forests-access-management/assets/23131735/f7456c88-6d9b-40e4-812b-a291bcb31a08">

Reply:
Hi Catherine, 
The exception handler is for FastAPI (router), it only catches CRUD exception when there is exception propagates to the router or throw from CRUD (if CRUD does not handle it).
However, on the second thought, in the future when granting access (CRUD) interacts/uses the new gc_notify class, then it will become relevant. So leaving it there is fine.

The second point that I mentioned is because I see this and is not clear to me (maybe only I am confused) from the GC Notify instruction; want to see if you are aware if it needs some "**status**" checking to really make sure it sends (or maybe we don't care if it really sends...). So, if we care, possibly maybe we need to handle error specifically in the class, like retry...?
![image](https://github.com/bcgov/nr-forests-access-management/assets/81595625/93bc1715-f6eb-4fb2-a3cf-72561ddae431)
@basilv

Reply:
I tested with invalid request, and it can catch the error successfully. It's hard to test the 5xx case though.

Reply:
I got Ian's point. Wonder how we'd like to handle the error case, do we want to retry to send the email? or what we'll do?

Reply:
I got Ian's point. Wondering how we'll handle the error case, if we want to try to resend the email? or what else we'll do?

Reply:
Created a new ticket for integration in frontend and the error handling #799